supply/lib/supply/client.rb

```diff
@@ -78,8 +78,10 @@ def initialize(service_account_json: nil, params: nil)
 
     def call_google_api
       yield if block_given?
-    rescue Google::Apis::ClientError => e
-      UI.user_error!("Google Api Error: #{e.message}")
+    rescue Google::Apis::Error => e
+      error = JSON.parse(e.body)
+      message = error["error"]["message"]
+      UI.user_error!("Google Api Error: #{e.message} - #{message}")
```

Comment:
I would probably actually do something a bit safer here since we are inside of a `rescue` and don't want to cause another error. Maybe something like this? 👇 (but feel free to make prettier)
```rb
error = JSON.parse(e.body) rescue nil
UI.verbose("Error parsing Google::Apis::Error body") unless error
error ||= {}

message = (error["error"] || {})["message"]
message = " - #{message}" if message
UI.user_error!("Google Api Error: #{e.message}#{message}")
```

Note: this is untested ^